### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by ZuhRa-HashiMi**
Set up associations between models.
Write SQL queries with Active Record.

### CODE REVIEW

**Review by Davidosky007 — COMMENTED** [review]
Hi @ZuhRa-HashiMi 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have MIT link in the readme file  

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Davidosky007 — COMMENTED** [review]
Hi @ZuhRa-HashiMi 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have MIT link in the readme file  

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class Post < ApplicationRecord
+end
```
Comment (codepantha) [inline app/models/post.rb]: - [ ] I noticed that you have not set up the associations for the `Post` model. Please try to set it up 👍 
- [ ] Please implement the following requirements on the `Post` model 👇 
image

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+class User < ApplicationRecord
+end
```
Comment (codepantha) [inline app/models/user.rb]: - [ ] I noticed that you have not set up the associations for the `User` model. Please try to set it up 👍 
- [ ] Please create a method that returns the 3 most recent posts for a given user.

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+class Like < ApplicationRecord
+end
```
Comment (codepantha) [inline app/models/like.rb]: - [ ] I noticed that you have not set up the associations for the `Like` model. Please try to set it up 👍
- [ ] Please create a method that updates the likes counter for a post.

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+class Comment < ApplicationRecord
+end
```
Comment (codepantha) [inline app/models/comment.rb]: - [ ] I noticed that you have not set up the associations for the `Comment` model. Please try to set it up 👍
- [ ] Please create a method that updates the comments counter for a post.

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @ZuhRa-HashiMi 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Used GitFlow
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

- [ ] Please set up the associations between models correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by alexander16108 — APPROVED** [review]
Hi @ZuhRa-HashiMi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✔️ Descriptive PR
✔️ Used GitFlow
✔️ No linter **errors**

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**